src/librustc_incremental/calculate_svh/mod.rs

```diff
@@ -149,19 +149,27 @@ impl<'a, 'tcx> HashItemsVisitor<'a, 'tcx> {
     {
         assert!(def_id.is_local());
         debug!("HashItemsVisitor::calculate(def_id={:?})", def_id);
+        self.calculate_def_hash(DepNode::Hir(def_id), false, &mut walk_op);
+        self.calculate_def_hash(DepNode::HirBody(def_id), true, &mut walk_op);
```

Comment:
If we were using `DepNode::Hir` for everything, then this code would not (I think) have to change; we would probably just never walk bodies in the hasher. Instead, we would call `calculate_def_id` on the bodies of fns as we traversed.